lib/transaction/transaction.js

```diff
@@ -49,21 +49,27 @@ function Transaction(serialized) {
   this._inputAmount = undefined;
   this._outputAmount = undefined;
 
-  if (serialized) {
-    if (serialized instanceof Transaction) {
-      return Transaction.shallowCopy(serialized);
-    } else if (JSUtil.isHexa(serialized)) {
-      this.fromString(serialized);
-    } else if (BufferUtil.isBuffer(serialized)) {
-      this.fromBuffer(serialized);
-    } else if (_.isObject(serialized)) {
-      this.fromObject(serialized);
+  try {
+    if (serialized) {
+      if (serialized instanceof Transaction) {
+        return Transaction.shallowCopy(serialized);
+      } else if (JSUtil.isHexa(serialized)) {
+        this.fromString(serialized);
+      } else if (BufferUtil.isBuffer(serialized)) {
+        this.fromBuffer(serialized);
+      } else if (_.isObject(serialized)) {
+        this.fromObject(serialized);
+      } else {
+        throw new errors.InvalidArgument('Must provide an object or string to deserialize a transaction');
+      }
     } else {
-      throw new errors.InvalidArgument('Must provide an object or string to deserialize a transaction');
+      this._newTransaction();
     }
-  } else {
-    this._newTransaction();
+  } catch (e){
+    console.error(`Failed transaction parsing with error ${e.message}. Transaction parameter: ${serialized}`);
```

Comment:
It's kinda wired to catch the error thrown on line 63 here. Probably you need to restructure the code a bit?

Reply:
I am not sure, do you have any idea on how you would want to restructure it ?

Reply:
Me too, honestly:))